carddav/carddav.go

```diff
@@ -99,3 +99,17 @@ type AddressObject struct {
 	ETag    string
 	Card    vcard.Card
 }
+
+//SyncQuery is the query struct represents a sync-collection request
+type SyncQuery struct {
+	DataRequest AddressDataRequest
+	SyncToken   string
+	Limit       int // <= 0 means unlimited
+}
+
+//SyncResponse contains the returned sync-token for next time
+type SyncResponse struct {
+	SyncToken string
+	Updated   []AddressObject
+	Deleted   []AddressObject
```

Comment:
Deleted address objects will only have a href, no actual data. So it would make sense to make this a list of paths instead (`[]string`).